src/util.rs

```diff
@@ -0,0 +1,42 @@
+pub fn layout_from_index(index: &[(u32, u32)], size: u32) -> Vec<(u32, u32)> {
+	if index.is_empty() {
+		return vec![(0, size)];
+	}
+
+	let (app_ids, offsets): (Vec<_>, Vec<_>) = index.iter().cloned().unzip();
+	// Prepend app_id zero
+	let mut app_ids_ext = vec![0];
+	app_ids_ext.extend(app_ids);
+
+	// Prepend offset 0 for app_id 0
+	let mut offsets_ext = vec![0];
+	offsets_ext.extend(offsets);
+
+	let mut sizes = offsets_ext[0..offsets_ext.len() - 1]
```

Comment:
Well thought off :)

Reply:
👍
We have to replace layouts with offsets and remove this for good.